projects/plugins/social/src/js/editor.js

```diff
@@ -27,27 +30,37 @@ domReady( () => {
 } );
 
 registerPlugin( 'jetpack-social', {
-	render: () => (
+	render: () => <JetpackSocialSidebar />,
+} );
+
+const JetpackSocialSidebar = () => {
+	const [ isModalOpened, setIsModalOpened ] = useState( false );
+
+	const openModal = useCallback( () => setIsModalOpened( true ), [] );
+	const closeModal = useCallback( () => setIsModalOpened( false ), [] );
+
+	return (
 		<PostTypeSupportCheck supportKeys="publicize">
+			{ isModalOpened && <SocialPreviewsModal onClose={ closeModal } /> }
```

Comment:
Does the modal need to rendered here? If not, then could we render the modal and panel in a `SocialPreviews` component, and have the state handled in there?

Having said that, I think we would need to pass in the PluginSidebar component, which could get messy. Perhaps we could create a custom `useSocialPreviews` hook that could return the two components. Something for another PR!

Reply:
Yeah, it's all kind of intertwined in the current setup. I like the idea of making a custom hook in a new PR.

Reply:
Yeah I think there's something better we can do here, but it's not for this PR!